src/confighttp.cpp

```diff
@@ -142,21 +176,78 @@ namespace confighttp {
     return true;
   }
 
+  /**
+   * @brief Send a 404 Not Found response.
+   * @param response The HTTP response object.
+   * @param request The HTTP request object.
+   */
   void
-  not_found(resp_https_t response, req_https_t request) {
+  not_found(resp_https_t response, [[maybe_unused]] req_https_t request) {
+    constexpr SimpleWeb::StatusCode code = SimpleWeb::StatusCode::client_error_not_found;
+
     pt::ptree tree;
-    tree.put("root.<xmlattr>.status_code", 404);
+    tree.put("status_code", static_cast<int>(code));
+    tree.put("error", "Not Found");
 
     std::ostringstream data;
+    pt::write_json(data, tree);
 
-    pt::write_xml(data, tree);
-    response->write(data.str());
+    SimpleWeb::CaseInsensitiveMultimap headers;
+    headers.emplace("Content-Type", "application/json");
+
+    response->write(SimpleWeb::StatusCode::client_error_not_found, data.str(), headers);
+  }
 
-    *response << "HTTP/1.1 404 NOT FOUND\r\n"
-              << data.str();
+  /**
+   * @brief Send a 400 Bad Request response.
+   * @param response The HTTP response object.
+   * @param request The HTTP request object.
+   */
+  void
+  bad_request_default(resp_https_t response, [[maybe_unused]] req_https_t request) {
+    constexpr SimpleWeb::StatusCode code = SimpleWeb::StatusCode::client_error_bad_request;
+
+    pt::ptree tree;
+    tree.put("status_code", static_cast<int>(code));
+    tree.put("error", "Invalid Request");
+
+    std::ostringstream data;
+    pt::write_json(data, tree);
+
+    SimpleWeb::CaseInsensitiveMultimap headers;
+    headers.emplace("Content-Type", "application/json");
+
+    response->write(SimpleWeb::StatusCode::client_error_bad_request, data.str(), headers);
   }
 
   /**
+   * @brief Send a 400 Bad Request response.
+   * @param response The HTTP response object.
+   * @param request The HTTP request object.
+   * @param error_message The error message to include in the response.
+   */
+  void
+  bad_request(resp_https_t response, [[maybe_unused]] req_https_t request, const std::string &error_message) {
```

Comment:
```suggestion
  bad_request(resp_https_t response, [[maybe_unused]] req_https_t request, const std::string &error_message = "Invalid Request") {
```

If we do this, could we perhaps get rid of `bad_request_default`?